measurement/plugins/download_speed/measurements.py

```diff
@@ -219,16 +219,19 @@ def _get_wget_results(self, url, download_timeout):
             return self._get_wget_error("wget-regex", url, traceback=wget_out.stderr)
 
         try:
-            storage_unit = NetworkUnit(
-                match_data.get("download_unit").replace("MB/s", "Mbit/s")
-            )
-        except ValueError:
+            download_rate_unit_map = {
+                "KB/s": NetworkUnit("Kibit/s"),
+                "MB/s": NetworkUnit("Mibit/s"),
+            }
+            download_rate_unit = download_rate_unit_map[match_data.get("download_unit")]
+        except KeyError:
             return self._get_wget_error(
-                "wget-storage-unit", url, traceback=wget_out.stderr
+                "wget-download-unit", url, traceback=wget_out.stderr
             )
 
         try:
-            download_rate = float(match_data.get("download_rate"))
+            # NOTE: wget returns download rate in [K|M]B/s. Convert to [K|M]ibit/s.
+            download_rate = float(match_data.get("download_rate")) * 8
```

Comment:
Are we actually better off using `Decimal` than float here now we are doing any manipulation on it due to exactness issues 

More details at: https://docs.python.org/2/library/decimal.html

Reply:
Good point. However that would mean a change to the `DownloadSpeedMeasurementResult` interface which might not be desirable alongside the number of bugs being squashed here. The numbers we're dealing with are small so I'm reasonably confident we don't have serious precision issues.

Reply:
As discussed in slack this is about anticipated behaviour and formatting issues that can occur. It has been discussed as we are aware of the edge cases and it is considered acceptable risk for the time being